processing/src/main/java/org/apache/druid/segment/column/IndexedStringDictionaryEncodedStringValueIndex.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment.column;
+
+import org.apache.druid.collections.bitmap.BitmapFactory;
+import org.apache.druid.collections.bitmap.ImmutableBitmap;
+import org.apache.druid.segment.data.Indexed;
+
+import javax.annotation.Nullable;
+
+public final class IndexedStringDictionaryEncodedStringValueIndex<T extends Indexed<String>>
```

Comment:
fwiw I didn't really introduce anything new here, I just moved all of these classes out of `DictionaryEncodedStringIndexSupplier`, renamed them, and added a generic parameter for the type of dictionary. I can try to make some adjustments, though not entirely sure what the be arrangement is. The main reason to have a base class was to not copy the `getBitmap` method everywhere since all of the implementations shared that in common, but I suppose there is another way that could be shared, will think about it.

Reply:
i removed the base class since really only 2 things needed to share the method, and the other thing using it (range index impl) didn't actually need to call this method because its ids are always values that exist (because is a range of ids that are in the dictionary)